### PR TITLE
Implemented I2C IMU on ESP-IDF

### PR DESCRIPTION
**Pull request by AugustDG**
* Supports MPU6050
* Uses an event-based system
* Runs on ESP High Precision Timer
* Uses new ESP I2C Driver 
* Modularizes into the following components:
  * I2CDev
  * MPU6050
  * IMU

### CODE REVIEW

File: setup.sh
Comment (Shamiivan) [inline setup.sh:1]: Can be different depending on the device. Maybe it's best to containerize it?

**Comment by Shamiivan** [comment]
Not sure, why but first time i built the package, the dependecies.lock was pointing nyquist/* as a path giving an error. We can either warn about that or put the file in gitignore.

**Comment by Shamiivan** [comment]
main.c should be removed.

**Comment by Shamiivan** [comment]
Does the imu data auto refresh or is there function call we have to make?

**Comment by AugustDG** [comment]
> Not sure, why but first time i built the package, the dependecies.lock was pointing nyquist/* as a path giving an error. We can either warn about that or put the file in gitignore.

Gitignore is the way to go!

**Comment by AugustDG** [comment]
> Does the imu data auto refresh or is there function call we have to make?

Auto-refresh based on the period given. I opted for a design that was consistent, instead of interrupt based, so we could have more control.

**Review by Shamiivan — APPROVED** [review]
Look good to me!

File: setup.sh
Comment (AugustDG) [reply setup.sh:1]: The build setup? Yes, absolutely! Hopefully ill have time today to do it :)

**Pull request merged**